Model/VatNumber/Config/Converter.php

```diff
@@ -31,35 +31,40 @@ public function __construct(BooleanUtils $booleanUtils)
     }
 
     /**
-     * Convert config.
+     * Convert config creating an assoc array with the country code as key
+     * and the pattern and other properties as value
      *
      * @param DOMDocument $source
      * @return array
+     * @throws \InvalidArgumentException
      */
     public function convert($source): array
     {
         $result = [];
 
-        /** @var DOMNode $vatNumberNode */
+        /** @var \DOMNode $vatNumberNode */
         foreach ($source->documentElement->childNodes as $vatNumberNode) {
-            if ($vatNumberNode->nodeType !== XML_ELEMENT_NODE
-                || !$this->booleanUtils->toBoolean($vatNumberNode->attributes->getNamedItem('active')->nodeValue ?? true)) {
-                continue;
-            }
-            $pattern = $vatNumberNode->textContent;
-            $id = $vatNumberNode->attributes->getNamedItem('id')->nodeValue;
-            $countryCode = strtoupper($vatNumberNode->attributes->getNamedItem('countryCode')->nodeValue);
-            $example = $vatNumberNode->attributes->getNamedItem('example')->nodeValue;
+            $countryCode = $vatNumberNode->attributes->getNamedItem('countryCode')->nodeValue;
+            $result[strtoupper($countryCode)] = $this->stripAndValidatePattern($vatNumberNode->textContent);
+        }
 
-            $pattern = trim($pattern);
+        return $result;
+    }
 
-            $result[$countryCode] = [
-                'id' => $id,
-                'pattern' => $pattern,
-                'example' => $example,
-            ];
+    /**
+     * Remove unwanted characters and validate regex
+     * @param string $pattern
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    public function stripAndValidatePattern(string $pattern): string
+    {
+        $pattern = trim($vatNumberNode->textContent);
```

Comment:
Nice one, maybe it's possible to add regex validation to the xsd?

Reply:
not too sure about that since regex patterns are quite complex with start and ends, escaping etc